api/v1/client.go

```diff
@@ -845,6 +849,173 @@ func (c *storageClient) Account(ctx context.Context, r *http.Request) (*Account,
 	return &a, nil
 }
 
+// Delegations returns a list of delegations.
+func (c *storageClient) Delegations(ctx context.Context, r *http.Request) (*DelegationList, error) {
+	chainID, ok := ctx.Value(ChainIDContextKey).(string)
+	if !ok {
+		return nil, common.ErrBadChainID
+	}
+
+	qb := NewQueryBuilder(fmt.Sprintf(`
+			SELECT delegatee, shares, escrow_balance_active, escrow_total_shares_active
+				FROM %s.delegations
+				JOIN %s.accounts ON %s.delegations.delegatee = %s.accounts.address
+			`,
+		chainID, chainID, chainID, chainID), c.db)
+
+	params := r.URL.Query()
+
+	var filters []string
+	for param, condition := range map[string]string{
+		"delegator": "delegator = %s",
+	} {
+		if v := params.Get(param); v != "" {
+			filters = append(filters, fmt.Sprintf(condition, v))
+		}
+	}
+	if err := qb.AddFilters(ctx, filters); err != nil {
+		c.logger.Info("filtering failed",
+			"request_id", ctx.Value(RequestIDContextKey),
+			"err", err.Error(),
+		)
+		return nil, common.ErrBadRequest
+	}
+
+	pagination, err := common.NewPagination(r)
+	if err != nil {
+		c.logger.Info("pagination failed",
+			"request_id", ctx.Value(RequestIDContextKey),
+			"err", err.Error(),
+		)
+		return nil, common.ErrBadRequest
+	}
+	if err = qb.AddPagination(ctx, pagination); err != nil {
+		c.logger.Info("pagination add failed",
+			"request_id", ctx.Value(RequestIDContextKey),
+			"err", err.Error(),
+		)
+		return nil, common.ErrBadRequest
+	}
+
+	rows, err := c.db.Query(ctx, qb.String())
+	if err != nil {
+		c.logger.Info("query failed",
+			"request_id", ctx.Value(RequestIDContextKey),
+			"err", err.Error(),
+		)
+		return nil, common.ErrStorageError
+	}
+	defer rows.Close()
+
+	var ds DelegationList
+	for rows.Next() {
+		var d Delegation
+		var escrowBalanceActive uint64
+		var escrowTotalSharesActive uint64
+		if err := rows.Scan(
+			&d.ValidatorAddress,
+			&d.Shares,
+			&escrowBalanceActive,
+			&escrowTotalSharesActive,
+		); err != nil {
+			c.logger.Info("row scan failed",
+				"request_id", ctx.Value(RequestIDContextKey),
+				"err", err.Error(),
+			)
+			return nil, common.ErrStorageError
+		}
+
+		d.Amount = float64(d.Shares*escrowBalanceActive) / float64(escrowTotalSharesActive)
+
+		ds.Delegations = append(ds.Delegations, d)
+	}
+
+	return &ds, nil
+}
+
+// DebondingDelegations returns a list of debonding delegations.
+func (c *storageClient) DebondingDelegations(ctx context.Context, r *http.Request) (*DebondingDelegationList, error) {
+	chainID, ok := ctx.Value(ChainIDContextKey).(string)
+	if !ok {
+		return nil, common.ErrBadChainID
+	}
+
+	qb := NewQueryBuilder(fmt.Sprintf(`
+			SELECT delegatee, shares, debond_end, escrow_balance_debonding, escrow_total_shares_debonding
+				FROM %s.debonding_delegations
+				JOIN %s.accounts ON %s.debonding_delegations.delegatee = %s.accounts.address
+				`,
```

Comment:
```suggestion
			SELECT delegatee, shares, debond_end, escrow_balance_debonding, escrow_total_shares_debonding
				FROM %s.debonding_delegations
				WHERE delegator = $1::text;
				`,
```
likewise here, and remove the `filters` below.

Reply:
Yeah, that is better.